test-utils/src/main/java/com/criteo/publisher/concurrent/MultiThreadResource.java

```diff
@@ -54,6 +56,10 @@ public void setUp() {
 
     dependencyProvider.inject(Executor.class, trackingCommandsExecutor);
     dependencyProvider.inject(AsyncResources.class, trackingCommandsExecutor.asAsyncResources());
+    if (!InstrumentationUtil.isRunningInInstrumentationTest()) {
+      // Main Lopper is not available in unit tests, so just mock RunOnUiThreadExecutor to avoid NullPointerException
+      dependencyProvider.inject(RunOnUiThreadExecutor.class, mock(RunOnUiThreadExecutor.class));
```

Comment:
You might want to have a real executor rather than a mock that will do nothing

Reply:
As UI thread is a unique thread, you could use the single thread executor from java (https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/Executors.html#newSingleThreadExecutor()) and wrap it inside a TrackingCommandsExecutor which would be waited for during the tear down

Reply:
This is a bit more complicated to do since we need an instance of RunOnUiThreadExecutor. We could subclass RunOnUiThreadExecutor with class that has TrackingCommandsExecutor in constuctor and delegate all method calls

Reply:
Alternative approarch is to use `DirectMockRunOnUiThreadExecutor` which will run code on thread we currently on.
Using separate thread adds some flakiness to unit tests

Reply:
Ok